docarray/helper.py

```diff
@@ -491,6 +491,42 @@ def _get_array_info(da: 'DocumentArray'):
     return is_homo, _nested_in, _nested_items, attr_counter, all_attrs_names
 
 
+def get_root_docs(da: 'DocumentArray', docs: 'DocumentArray'):
+    from docarray import DocumentArray
+
+    root_da_flat = da[...]
+    da = DocumentArray()
+    for doc in docs:
+        result = doc
+        while getattr(result, 'parent_id', None):
+            result = root_da_flat[result.parent_id]
+        da.append(result)
+    return da
+
+
+def check_root_id(da: 'DocumentArray', value: Union['Document', Sequence['Document']]):
+
+    from docarray import Document
+    from docarray.array.memory import DocumentArrayInMemory
+
+    if isinstance(value, Document):
+        value = [value]
+
+    if isinstance(da, DocumentArrayInMemory):
+        if not all([getattr(doc, 'parent_id', None) for doc in value]):
```

Comment:
this should look into `tags` and not check if the document itself has the root id, right?

Reply:
`DocumentArrayInMemory` can use `parent_id` recursively to get there root level Document, so I didn't store extra `root_id` information in tags for in-memory. [current code](https://github.com/docarray/docarray/pull/808/files#diff-d1f35e3f3298a94b90e3a354c069cf641edad1a1374551b14befde43666c370d)